cmd/internal/flowmode/cmd_run.go

```diff
@@ -334,6 +348,13 @@ func (fr *flowRun) Run(configFile string) error {
 		return err
 	}
 
+	// Setup config file watching
+	if fr.configAutoReload {
+		fw := util.NewFileWatcher(configFile, reload, fr.configPollInterval, fr.configForcePolling)
+		fw.Watch(l, ctx, &wg)
```

Comment:
Passing the WaitGroup to watch is leaking the implementation details of the caller to the shared function, which is a bit of a smell IMO. 

I'd suggest to change the details here a bit:

* Make `fw.Watch` a blocking call. 
* Call `fw.Watch` from a goroutine launched by the caller:

  ```go 
  wg.Add(1) 
  go func() {
    defer wg.Done()
    fw.Watch(l, ctx) 
  }()
  ```

This will allow util.NewFileWatcher to be used in scenarios where a WaitGroup isn't necessary.

Reply:
Great point - I'll correct